File: Problems/Algorithms/290. Word Pattern/word_pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        arr = s.split(" ")
        if len(pattern) != len(arr):
            return False

        bijection = {}
        for i in range(len(pattern)):
            ch = pattern[i]
            if ch not in bijection:
                if arr[i] in bijection.values():
                    return False
                bijection[ch] = arr[i]
            else:
                if arr[i] != bijection[ch]:
                    return False

        return True

File: Problems/Algorithms/290. Word Pattern/WordPattern.kt
class Solution {
    fun wordPattern(pattern: String, s: String): Boolean {
        val arr = s.split(" ")
        if (arr.size != pattern.length) {
            return false
        }
        
        val bijection = mutableMapOf<Char, String>()
        for (i in 0..pattern.length-1) {
            val ch = pattern.get(i)
            if (!bijection.containsKey(ch)) {
                if (bijection.containsValue(arr[i])) {
                    return false
                }
                bijection.put(ch, arr[i])
            } else {
                if (arr[i] != bijection[ch]) {
                    return false
                }
            }
        }
        
        return true
    }
}

